### PR TITLE
ci: set proxy and file chown

### PR DESCRIPTION
**Pull request by junruizh2021**
- Add proxy in build-kata-deploy and kata-kernel-builder dockerfile
- Add chown command for ${USER} before each checkout@v2 step

Fixes: #3465
Signed-off-by: junruizh2021 <junrui.zhang@intel.com>

### CODE REVIEW

**Review by amshinde — APPROVED** [review]

File: .github/workflows/add-issues-to-project.yaml
Comment (Jakob-Naucke) [inline .github/workflows/add-issues-to-project.yaml:38]: 1. I think this can be prettified a little with `$GITHUB_WORKSPACE`.
2. Honestly, I wonder why GHA is attempting `git clean` at all. Aren't those environments immediately thrown away anyhow?
3. Is this necessary for all the workflows? E.g. I don't see how we're adding anything to the workspace by root in this workflow here.

File: tools/packaging/static-build/kernel/Dockerfile
Comment (Jakob-Naucke) [inline tools/packaging/static-build/kernel/Dockerfile:11]: Isn't `$ARG` enough?

**Review by Jakob-Naucke — COMMENTED** [review]
thank you @junruizh2021 -- I have some comments

**Comment by Jakob-Naucke** [comment]
I'm also going to assume this would need a backport.

File: .github/workflows/add-issues-to-project.yaml
Comment (junruizh2021) [reply .github/workflows/add-issues-to-project.yaml]: 1. Thanks for the reminder, using $GITHUB_WORKSPACE to represent the path is more general.
2. This is the logic of GHA's CI yaml. Perhaps this can ensure that the test environment of each job is consistent with the upstream.
3. I refered to the configuration tutorial of "Add Github Action Runner" page, the CLI warns "Don't must use root privileges to start the runner", so my modifications here are all for non-root users.

File: tools/packaging/static-build/kernel/Dockerfile
Comment (junruizh2021) [reply tools/packaging/static-build/kernel/Dockerfile:11]: If no proxy is added, network problems will occur frequently in the intranet environment.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by bergwolf** [comment]
I wonder how `proxy` is going to help in the docker container network case, if the non-proxy container fail to connect to external server.  Is your goal to support running the tests in a private network field?

**Comment by junruizh2021** [comment]
> I wonder how `proxy` is going to help in the docker container network case, if the non-proxy container fail to connect to external server. Is your goal to support running the tests in a private network field?

@bergwolf Yes, I have deployed github action runner on the company's internal network. The host has set the company's proxy, but the docker inside the container is not configured with a proxy. What I want to do is to **add a proxy to the docker inside the container (build-kata-deploy)** to make sure that **docker build can run successfully**. If the docker of the build-kata-deploy container does not add an agent, a network error will occur during the docker build process (for example, when executing "apt-get update").

**Comment by junruizh2021** [comment]
> A head's up that may affect your PR.
> 
> In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub. It basically means that if your PR has been rebased or updated, the approvals given will be erased.
> 
> In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.
> 
> Thanks for your understanding!
> 
> Related issue: [kata-containers/community#249](https://github.com/kata-containers/community/issues/249)

@fidencio Thanks for your comment.

@Jakob-Naucke @amshinde Based on Jakob-Naucke's input, my PR has been updated. Please take your time to give any comments, thanks a lot.

File: .github/workflows/add-issues-to-project.yaml
Comment (Jakob-Naucke) [reply .github/workflows/add-issues-to-project.yaml]: 3. I'm really not in favour of `chown`ing in lots of places without necessity. Can you please limit this to the jobs where you saw issues with permissions?

File: tools/packaging/static-build/kernel/Dockerfile
Comment (Jakob-Naucke) [reply tools/packaging/static-build/kernel/Dockerfile:11]: No, I meant: You make Docker aware of `http_proxy` through `ARG http_proxy`, so is `ENV http_proxy ${http_proxy}` necessary?

Comment (junruizh2021) [reply tools/packaging/static-build/kernel/Dockerfile:11]: ARG exists at build time and can be used as a variable in the Dockerfile. ENV is an environment variable after the container is built and cannot be used as a parameter in the Dockerfile. I want to use http_proxy as the environment variable of the container. For convenience, both ARG and ENV are required.

File: .github/workflows/kata-deploy-push.yaml
Comment (junruizh2021) [reply .github/workflows/kata-deploy-push.yaml:63]: Yes, each checkout@v2 step cleans the working directory, so I added a "chown file access" action before each checkout@v2 step to ensure that the checkout action has permission to delete all files in the directory.

**Comment by jodh-intel** [comment]
Hi @junruizh2021 - please can you rebase to resolve the conflict?

**Review by Jakob-Naucke — COMMENTED** [review]
@junruizh2021 Please rebase your changes, do not run a merge or we'd have additional merge commits in our history.

File: tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile
Comment (liubin) [inline tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:48]: Can this be rewritten like this?

```bash
sh -c 'cat << EOF > ~/.docker/config.json
{
  "proxies": {
    "default": {
      "httpProxy": "${http_proxy}",
      "httpsProxy": "${https_proxy}",
      "noProxy": "localhost,127.0.0.1"
    }
  }
}
EOF'
```

Too many back quotas are not readable and easy to run into errors.

**Review by gkurz — CHANGES_REQUESTED** [review]
Hi @junruizh2021 !

This is addressing two unrelated topics IIUC. It would seem a lot better to split that into separate patches : one for the proxy and one for the chown. It would facilitate review, adoption and maintenance of each change independently from the other.

Also the commit description just paraphrases the changes, i.e. it doesn't bring much. It would be more useful to add some context so that one doesn't have to go through the PR/issue on github to understand the motivation of each change, e.g.
- proxy is needed to be able to run in a private network. Clarify the use of both ARG and ENV
- chown is needed because checkout@v2 does some cleaning and might face permission issues. Clarify which steps are affected.

**Comment by jodh-intel** [comment]
Any update on this @junruizh2021? I agree with @gkurz that this would be better as two commits. Also, there should not be a "merge commit" on this PR.

**Comment by junruizh2021** [comment]
> Hi @junruizh2021 !
> 
> This is addressing two unrelated topics IIUC. It would seem a lot better to split that into separate patches : one for the proxy and one for the chown. It would facilitate review, adoption and maintenance of each change independently from the other.
> 
> Also the commit description just paraphrases the changes, i.e. it doesn't bring much. It would be more useful to add some context so that one doesn't have to go through the PR/issue on github to understand the motivation of each change, e.g.
> 
> * proxy is needed to be able to run in a private network. Clarify the use of both ARG and ENV
> * chown is needed because checkout@v2 does some cleaning and might face permission issues. Clarify which steps are affected.

Thanks for your suggestion, I will consider closing this PR, then pull the latest branch and submit PRs for firewall proxy and permissions issues respectively.

**Comment by junruizh2021** [comment]
> Any update on this @junruizh2021? I agree with @gkurz that this would be better as two commits. Also, there should not be a "merge commit" on this PR.

Nope, I will close this PR. As suggested by @gkurz , I would pull the latest branch, submitting separate commits for firewall proxy and execute permissions.

**Comment by junruizh2021** [comment]
> Any update on this @junruizh2021? I agree with @gkurz that this would be better as two commits. Also, there should not be a "merge commit" on this PR.

Nope, I will close this PR. As suggested by gkurz @gkurz , I would pull the latest branch, submitting separate commits for firewall proxy and execute permissions.

> @junruizh2021 Any updates?

Sorry for my later reply, I will reconsider all comments. By submitting two commits, I will try to describe my ideas clearly and rebase the latest master branch.

File: tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile
Comment (junruizh2021) [reply tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:48]: @liubin Thanks, I will check it again.

**Pull request closed without merging**
